src/text-editor.coffee

```diff
@@ -993,17 +996,18 @@ class TextEditor extends Model
   # selections to create multiple single-line selections that cumulatively cover
   # the same original area.
   splitSelectionsIntoLines: ->
-    for selection in @getSelections()
-      range = selection.getBufferRange()
-      continue if range.isSingleLine()
+    @batch => @mergeIntersectingSelections =>
```

Comment:
Should the call to `mergeIntersectingSelections` always imply batching, similar to transactions?